app/src/main/java/ch/sdp/vibester/activity/GameSetupActivity.kt

```diff
@@ -30,67 +29,85 @@ import retrofit2.Response
                 findViewById<LinearLayout>(R.id.chooseGame).visibility = VISIBLE
                 findViewById<ConstraintLayout>(R.id.chooseGenre).visibility = GONE
-            }else if (findViewById<ConstraintLayout>(R.id.chooseDifficulty).visibility == VISIBLE){
+            } else if (findViewById<ConstraintLayout>(R.id.gameSetting).visibility == VISIBLE) {
                 findViewById<ConstraintLayout>(R.id.chooseGenre).visibility = VISIBLE
-                findViewById<ConstraintLayout>(R.id.chooseDifficulty).visibility = GONE
+                findViewById<ConstraintLayout>(R.id.gameSetting).visibility = GONE
             }
         }
+    }
 
-
-        val spinnerDifficulty: Spinner = findViewById(R.id.difficulty_spinner)
+    private fun setSpinnerListener(ctx: Context, spinnerId: Int, resourceId: Int) {
+        val spinner: Spinner = findViewById(spinnerId)
         val adapter = ArrayAdapter.createFromResource(
-            ctx,
-            R.array.difficulties_name,
-            android.R.layout.simple_spinner_item
+            ctx, resourceId, android.R.layout.simple_spinner_item
         )
         adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
-        spinnerDifficulty.adapter = adapter
-        spinnerDifficulty.onItemSelectedListener = this
+        spinner.adapter = adapter
+        spinner.onItemSelectedListener = this
     }
 
     override fun onItemSelected(parent: AdapterView<*>, view: View?, position: Int, id: Long) {
-        difficulty = parent.getItemAtPosition(position).toString()
-        when(difficulty) {
-            "Easy"      -> setDifficultyText(R.string.difficulty_easy)
-            "Medium"    -> setDifficultyText(R.string.difficulty_medium)
-            "Hard"      -> setDifficultyText(R.string.difficulty_hard)
+        if (parent.id == R.id.difficulty_spinner) {
+            difficulty = parent.getItemAtPosition(position).toString()
+            when (difficulty) {
+                "Easy"      -> setDifficultyText(R.string.difficulty_easy)
+                "Medium"    -> setDifficultyText(R.string.difficulty_medium)
+                "Hard"      -> setDifficultyText(R.string.difficulty_hard)
+            }
+        } else if (parent.id == R.id.size_spinner) {
+            gameSize = parent.getItemAtPosition(position).toString()
+            when (gameSize) {
+                "One" -> gameManager.setGameSize(1)
+                "Two" -> gameManager.setGameSize(2)
+                "Three" -> gameManager.setGameSize(3)
+                "Four" -> gameManager.setGameSize(4)
+                "Five" -> gameManager.setGameSize(5)
+                "Six" -> gameManager.setGameSize(6)
+                "Seven" -> gameManager.setGameSize(7)
+                "Eight" -> gameManager.setGameSize(8)
+                "Nine" -> gameManager.setGameSize(9)
+                "Ten" -> gameManager.setGameSize(10)
```

Comment:
We may use string.xml for storing these. If one day we want to translate the app, it would be necessary to do it